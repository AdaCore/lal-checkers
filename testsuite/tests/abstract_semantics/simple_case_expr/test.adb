procedure Ex1 is
   x : Integer;
   y : Integer;
begin
   y := case x is
        when 2 .. 4 | 6 .. 8 => 42,
        when 314 => 1592,
        when others => x;
end Ex1;
