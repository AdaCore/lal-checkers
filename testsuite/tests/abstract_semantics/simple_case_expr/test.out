{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign3", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign4", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assign5", 
          "assume0", 
          "assume1", 
          "assume2", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "split_join0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assume0", 
          "assume1", 
          "assume2", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign3", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign4", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign5", 
          "assume0", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[42, 42]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign3", 
          "assign5", 
          "assume1", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[1592, 1592]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign4", 
          "assign5", 
          "assume2", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "split_join0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assume0", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign3", 
          "assume1", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[314, 314]", 
          "y": "[-2147483648, 2147483647]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign4", 
          "assume2", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "x": "[-2147483648, 2147483647]", 
          "y": "[-2147483648, 2147483647]"
        }
      }
    ]
  }
}
